pkg/controller/gitopsconfig/controller.go

```diff
@@ -244,19 +246,19 @@ func (r *Reconciler) createJob(jobtype string, instance *gitopsv1alpha1.GitOpsCo
 	job, err := util.CreateJob(mergedata)
 	if err != nil {
 		log.Error(err, "unable to create job manifest from merge data", "mergedata", mergedata)
-		return reconcile.Result{}, err
+		return reconcile.Result{}, xerrors.Errorf("unable to create job manifest from merge data %q: %w", mergedata, err)
 	}
 	err = controllerutil.SetControllerReference(instance, &job, r.scheme)
 	if err != nil {
 		log.Error(err, "unable to the owner for job", "job", job)
-		return reconcile.Result{}, err
+		return reconcile.Result{}, xerrors.Errorf("unable to the owner for job %q: %w", job, err)
```

Comment:
1. the sentence doesn't seem to gramatically make sense to me; can you try to understand or ask what the original author meant here, and improve this message in both lines above?
2. I am not sure %q will print something sensible for whole "job" object
3. In a similar vein, please review all the remaining  uses of `%q` in this PR, to make sure they actually print something sensible and useful, while also not dumping whole huge object; please let me know afterwards, I'll come back to reviewing this PR then